client/src/pages/course/student/dashboard.tsx

```diff
@@ -43,8 +50,10 @@ function Page(props: CoursePageProps) {
   const [studentSummary, setStudentSummary] = useState({} as StudentSummary);
   const [repositoryUrl, setRepositoryUrl] = useState('');
   const [courseTasks, setCourseTasks] = useState<CourseTaskDto[]>([]);
-  const [tasksDetail, setTasksDetail] = useState<StudentTasksDetail[]>([]);
   const [nextEvents, setNextEvent] = useState([] as CourseEvent[]);
+  const [tasksByStatus, setTasksByStatus] = useState(
```

Comment:
useState can accept generic type:
```
useState<Record<CourseScheduleItemDtoStatusEnum, CourseScheduleItemDto[]>>()
```
Maybe it's even better to extract it to type alias

Reply:
this doesn't work well with record type

Reply:
this doesn't work well with record type - ie you get type error

Reply:
Yes, because in general using objects with useState is not good idea 🙂 It's recommended to refactor it to useReducer, but it's up to you

Reply:
`in general using objects with useState is not good idea` could u please share any references to such recommendations. I hear this for the first time and if I am using outdated approach would be great to know about it

Reply:
It's here https://reactjs.org/docs/hooks-reference.html#usereducer

> useReducer is usually preferable to useState when you have complex state logic that involves multiple sub-values or when the next state depends on the previous one.

Reply:
right, but this is not about not using objects in usestate. this is absolutely different thing: about changing the data(mutating, merging and etc), not storing. Storing object absolutely legal and recommended. the problem is with typescript, that `record<enum, anything>` doesn't work well with usestate typing.

Reply:
Yeah, sure. As I said it's up to you